06-recon-enumeration/README.md

```diff
@@ -0,0 +1,122 @@
+
+There are also many different types of payloads you can use [[7]](https://portswigger.net/burp/documentation/desktop/tools/intruder/payloads/types), from specifying your own list of words to generating random bytes. You can find lists of popular credentials online, for instance, here it is a repo with lists of most used passwords [[8]](https://github.com/danielmiessler/SecLists/tree/master/Passwords).
+
+
+## Brute Force Active Directory
+
+Let's say we have the target server **http://192.168.1.224/** and we want to discover hidden files, directories or other resources there. Manually, we would make multiple requests like _http://192.168.1.224/docs_, _http://192.168.1.224/config.php_ etc. or whatever we imagine might find and see if we get a 404 Not Found response or not. Luckily, there are command line tools and predefined wordlists in Kali (**/usr/share/wordlists/**) doing exactly this for us.
+
+## DIRB
+DIRB [[9]](https://tools.kali.org/web-applications/dirb) is a Web Content Scanner, a Kali built in tool. It looks for existing (and/or hidden) Web Objects. It basically works by launching a dictionary based attack against a web server and analyzing the response.
+DIRB comes with a set of preconfigured attack wordlists for easy usage but you can use your custom wordlists. For each filename, it check the  existence on the webserver and returns the results which do not give a 404 Not Found response.
+
+Usage example: `./dirb <url_base> [<wordlist_file(s)>] [options]`
+
+![DIRB example](https://github.com/hexcellents/sss-web/blob/master/06-recon-enumeration/support/dirb_example.png)
+
+The output lines with the results found (not 404) start with a `+` and give details about status code and page size.
+
+You can read the documentation if you want to specify custom options, like custom file extensions to look for.
+
+## Similar tools
+* DirBuster [[10]](https://tools.kali.org/web-applications/dirbuster) - Kali built in, written in Java and the only one with a GUI and not a CLI
+* GoBuster [[11]](https://tools.kali.org/web-applications/gobuster) - Kali built in, written in Go, also run in command line, but with more configurable options, like setting cookies or the User-Agent
+* wfuzz [[12]](https://github.com/xmendez/wfuzz) - available on GitHub, written in Python, has a lot of options, can be easily installed with pip 
+* ffuf [[13]](https://github.com/ffuf/ffuf) - also on GitHub, written in Go, has the option to mutate the files found
+
+## Other techniques
+
+* Extracting user names using email ID's
+* Extract information using the default password         
+* Extract user names using SNMP
+* Extract user groups from Windows
+* Extract information using DNS Zone transfer
+
+Read more:
+https://www.knowledgehut.com/blog/security/enumeration-in-ethical-hacking
+https://www.greycampus.com/opencampus/ethical-hacking/enumeration-and-its-types
+
```

Comment:
There should be a session dubbed `## Challenges` or `## Tasks` or `## Exercises` or `## Practice` or `## Practical Activities`.